src/ripple/core/tests/Config.test.cpp

```diff
@@ -276,11 +276,14 @@ nHUhG1PgAG8H8myUENypM35JgfqXAKNQvRVVAFDRzJrny5eZN8d5
 nHBu9PTL9dn2GuZtdW4U2WzBwffyX9qsQCd9CNU4Z5YG3PQfViM8
 nHUPDdcdb2Y5DZAJne4c2iabFuAP3F34xZUgYQT2NH7qfkdapgnz
 
-[validation_quorum]
 %1%
+
 )rippleConfig");
 
-    return boost::str (configContentsTemplate % quorum);
+    std::string quorumSection =
+        quorum.empty () ? "" : "[validation_quorum]\n" + quorum;
+    return boost::str (
+        configContentsTemplate % quorumSection);
```

Comment:
Instead of `std::string quorum`, you could
1. Take a `boost::optional<int> quorum`.
2. initialize `quorumSection = quorum ? "[validation_quorum\n" + to_string(*quorum) : "";`
3. Change the `ValidatorsTxtGuard` ctor to pass the `boost::optional<int>` through.
4. Change the callers to pass a straight `int` when there's a value or `boost::none` when there isn't.

All this really does is slightly simplify the callers, so I'm treating it as completely optional.